File: api/javer/src/main/java/com/uuhnaut69/javer/repository/DishRepository.java
package com.uuhnaut69.javer.repository;

import com.uuhnaut69.javer.model.Dish;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface DishRepository extends JpaRepository<Dish, UUID> {

  @EntityGraph(attributePaths = {"restaurant"})
  List<Dish> findAllByRestaurantId(UUID restaurantId);
}

File: api/kover/src/main/kotlin/com/uuhnaut69/kover/repository/DishRepository.kt
package com.uuhnaut69.kover.repository

import com.uuhnaut69.kover.model.Dish
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface DishRepository : JpaRepository<Dish, UUID> {

    @EntityGraph(attributePaths = ["restaurant"])
    fun findAllByRestaurantId(restaurantId: UUID): List<Dish>
}